File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/data/DataStore.java
package org.kevoree.modeling.api.data;

import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.events.ModelElementListener;
import org.kevoree.modeling.api.events.ModelEvent;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 05/11/2013
 * Time: 11:30
 */

public interface DataStore {

    public void get(String key, Callback<String> callback, Callback<Exception> error);

    public void put(String key, String value, Callback<Boolean> callback, Callback<Exception> error);

    public void remove(String key, Callback<Boolean> callback, Callback<Exception> error);

    public void commit(Callback<String> callback, Callback<Exception> error);

    public void close(Callback<String> callback, Callback<Exception> error);

    public void notify(ModelEvent event);

    public void register(ModelElementListener listener, long from, long to, String path);

    public void unregister(ModelElementListener listener);

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/persistence/DataStore.kt
package org.kevoree.modeling.api.persistence

import org.kevoree.modeling.api.events.ModelEvent
import org.kevoree.modeling.api.events.ModelElementListener
import org.kevoree.modeling.api.Callback

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 05/11/2013
 * Time: 11:30
 */

public trait DataStore {

    fun put(key: String, value: String, callback : Callback<String>, error : Callback<Exception>)

    fun get(key: String, callback : Callback<String>, error : Callback<Exception>)

    fun remove(key: String, callback : Callback<String>, error : Callback<Exception>)

    fun commit(callback : Callback<String>, error : Callback<Exception>)

    fun close(callback : Callback<String>, error : Callback<Exception>)

    fun notify(event: ModelEvent)

    fun register(listener: org.kevoree.modeling.api.events.ModelElementListener, from: Long?, to: Long?, path: String)

    fun unregister(listener: org.kevoree.modeling.api.events.ModelElementListener)

}